### PR TITLE
th/settings-cleanup

### PR DESCRIPTION
**Pull request by thom311**
A branch with various cleanups of settings plugins, in particular to ifupdown plugin.

I think ifupdown plugin is widely used, and didn't get much attention in the past years. Let's invest some effort there.

### CODE REVIEW

**Comment by vbenes** [comment]
we have a crash here
https://faf.lab.eng.brq.redhat.com/faf/problems/bthash/?bth=597a3b075dbfac715f1dfba4a8c64c1e091d761e

**Comment by thom311** [comment]
@mbiebl, the ifupdown plugin didn't get much attention in the past years. I think there were things which could/should be improved.
This branch contains a larger cleanup of the plugin. We do have unit-tests, so I hope that no new bugs are introduced. But this is something to be aware of, when the next NetworkManager release goes to Debian.

Does Debian have additional CI/unit tests for the ifupdown plugin?

File: src/settings/plugins/ifupdown/nms-ifupdown-plugin.c [hunk 1]
```diff
@@ -252,14 +248,23 @@ get_connections (NMSettingsPlugin *plugin)
...
 		_LOGD ("get_connections: not connections due to managed=false");
 		return NULL;
 	}
 
-	_LOGD ("get_connections: %u connections", g_hash_table_size (priv->connections));
-	return _nm_utils_hash_values_to_slist (priv->connections);
+	g_hash_table_iter_init (&iter, priv->eni_ifaces);
+	while (g_hash_table_iter_next (&iter, NULL, &value)) {
+		if (value)
+			 list = g_slist_prepend (list, value);
```
Comment (bengal) [inline src/settings/plugins/ifupdown/nms-ifupdown-plugin.c]: Extra space before 'list'.

**Comment by thom311** [comment]
merged to master

**Comment by mbiebl** [comment]
> Does Debian have additional CI/unit tests for the ifupdown plugin?

We don't, unfortunately

**Comment by mbiebl** [comment]
Thanks for the heads-up. Will keep an eye on it when going to 1.13/1.14

**Pull request closed without merging**